src/main/java/de/rub/nds/modifiablevariable/ModifiableVariable.java

```diff
@@ -75,4 +75,21 @@ public void createRandomModificationAtRuntime() {
     public boolean containsAssertion() {
         return (assertEquals != null);
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o)
+            return true;
+        if (!(o instanceof ModifiableVariable))
+            return false;
+
+        ModifiableVariable<?> that = (ModifiableVariable<?>) o;
+
+        return modification != null ? modification.equals(that.modification) : that.modification == null;
```

Comment:
Can this really work? Since you did not impelemnt equals methods for the modifications. I also think that you do not need to implement Equals Methods. I think its enough if you compare the "getValue"?